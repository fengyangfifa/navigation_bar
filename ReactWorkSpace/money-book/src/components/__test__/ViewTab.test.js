import React from 'react'
import { shallow } from 'enzyme'
import ViewTab from '../ViewTab'
import { LIST_VIEW, CHART_VIEW } from '../../utility'


const props = {
  activeTab: LIST_VIEW,
  onTabChange: jest.fn()
}

let wrapper;
describe('test ViewTab component', () => {
  beforeEach(() => {
    wrapper = shallow(<ViewTab {...props}/>);
  });
  it('should render component to match snapshot', () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('测试第一个是不是列表模式和点击时传递的参数', () => {    
    const firstItem = wrapper.find('.nav-link').first();
    expect(firstItem.text()).toEqual('<Ionicon />列表模式');
    firstItem.simulate('click');
    expect(props.onTabChange).toHaveBeenCalledWith(LIST_VIEW);
  });
});

