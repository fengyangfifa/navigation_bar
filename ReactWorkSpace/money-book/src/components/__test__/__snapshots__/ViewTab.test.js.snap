// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component should render component to match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "nav nav-tabs nav-fill my-4",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "nav-item",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "nav-link active",
              "href": "#",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                  "fill": "#007bff",
                  "height": "25px",
                  "rotate": "0",
                  "viewBox": "0 0 1024 1024",
                  "width": "25px",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M224 96v736h-32v-608h-64v642.2c0 34 27.6 61.8 61.8 61.8h644.6c34 0 61.8-27.6 61.8-61.8v-770.2h-672.2zM320 224h256v32h-256v-32zM320 544h384v32h-384v-32zM800 736h-480v-32h480v32zM800 416h-480v-32h480v32z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": Object {
                  "data": "列表模式",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "fill": undefined,
                  "height": undefined,
                  "rotate": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "fill": undefined,
                  "height": undefined,
                  "rotate": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
              },
              Object {
                "data": "列表模式",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                    "fill": "#007bff",
                    "height": "25px",
                    "rotate": "0",
                    "viewBox": "0 0 1024 1024",
                    "width": "25px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M224 96v736h-32v-608h-64v642.2c0 34 27.6 61.8 61.8 61.8h644.6c34 0 61.8-27.6 61.8-61.8v-770.2h-672.2zM320 224h256v32h-256v-32zM320 544h384v32h-384v-32zM800 736h-480v-32h480v32zM800 416h-480v-32h480v32z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                },
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "nav-item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "nav-link",
                "href": "#",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                    "fill": "#007bff",
                    "height": "25px",
                    "rotate": "0",
                    "viewBox": "0 0 1024 1024",
                    "width": "25px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M544 128v448l-395.51 98.752c54.816 165.69 211.192 285.248 395.26 285.248 229.75 0 416.25-186.25 416.25-416s-186-416-416-416v0z M512 64h-11c-320 0-437 232.5-437 414.574 0 0 0.5 113.332 37.82 176.582l410.18-104.010v-487.146z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": Object {
                    "data": "图表模式",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                },
                Object {
                  "data": "图表模式",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                      "fill": "#007bff",
                      "height": "25px",
                      "rotate": "0",
                      "viewBox": "0 0 1024 1024",
                      "width": "25px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "d": "M544 128v448l-395.51 98.752c54.816 165.69 211.192 285.248 395.26 285.248 229.75 0 416.25-186.25 416.25-416s-186-416-416-416v0z M512 64h-11c-320 0-437 232.5-437 414.574 0 0 0.5 113.332 37.82 176.582l410.18-104.010v-487.146z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "fill": undefined,
                      "height": undefined,
                      "rotate": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "fill": undefined,
                      "height": undefined,
                      "rotate": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                    },
                  },
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "nav-item",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "nav-link",
              "href": "#",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                  "fill": "#007bff",
                  "height": "25px",
                  "rotate": "0",
                  "viewBox": "0 0 1024 1024",
                  "width": "25px",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M544 128v448l-395.51 98.752c54.816 165.69 211.192 285.248 395.26 285.248 229.75 0 416.25-186.25 416.25-416s-186-416-416-416v0z M512 64h-11c-320 0-437 232.5-437 414.574 0 0 0.5 113.332 37.82 176.582l410.18-104.010v-487.146z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": Object {
                  "data": "图表模式",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "fill": undefined,
                  "height": undefined,
                  "rotate": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "fill": undefined,
                  "height": undefined,
                  "rotate": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
              },
              Object {
                "data": "图表模式",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                    "fill": "#007bff",
                    "height": "25px",
                    "rotate": "0",
                    "viewBox": "0 0 1024 1024",
                    "width": "25px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M544 128v448l-395.51 98.752c54.816 165.69 211.192 285.248 395.26 285.248 229.75 0 416.25-186.25 416.25-416s-186-416-416-416v0z M512 64h-11c-320 0-437 232.5-437 414.574 0 0 0.5 113.332 37.82 176.582l410.18-104.010v-487.146z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                },
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "nav-item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "nav-link active",
                "href": "#",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                    "fill": "#007bff",
                    "height": "25px",
                    "rotate": "0",
                    "viewBox": "0 0 1024 1024",
                    "width": "25px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M224 96v736h-32v-608h-64v642.2c0 34 27.6 61.8 61.8 61.8h644.6c34 0 61.8-27.6 61.8-61.8v-770.2h-672.2zM320 224h256v32h-256v-32zM320 544h384v32h-384v-32zM800 736h-480v-32h480v32zM800 416h-480v-32h480v32z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": Object {
                    "data": "列表模式",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "rotate": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                },
                Object {
                  "data": "列表模式",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "rounded-circle mr-2 sc-bdVaJa fUuvxv",
                      "fill": "#007bff",
                      "height": "25px",
                      "rotate": "0",
                      "viewBox": "0 0 1024 1024",
                      "width": "25px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "d": "M224 96v736h-32v-608h-64v642.2c0 34 27.6 61.8 61.8 61.8h644.6c34 0 61.8-27.6 61.8-61.8v-770.2h-672.2zM320 224h256v32h-256v-32zM320 544h384v32h-384v-32zM800 736h-480v-32h480v32zM800 416h-480v-32h480v32z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "fill": undefined,
                      "height": undefined,
                      "rotate": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "fill": undefined,
                      "height": undefined,
                      "rotate": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                    },
                  },
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "ul",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
